### PR TITLE
Renaming SemanticConvention constants

### PR DESCRIPTION
**Pull request by alanwest**
This PR addresses a bit of a nit, though given that the `SemanticConventions` class is currently public, I personally feel it makes sense to conform to the [design guidelines](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions#capitalization-rules-for-identifiers) for capitalization of acronyms like `Http`, `Db`, `Rpc`, `Id`, and others.

That said, having discussed this with @cijothomas, there is also the question of whether this class even ought to be public. We could make it internal, though a fair number of projects depend on this class, so would we want to be doing something like this everywhere?:

```xml
<ItemGroup>
    <Compile Include="..\OpenTelemetry.Api\SemanticConventions.cs" />
</ItemGroup>
```

Or are there better ways of dealing with this? Or do we just leave the class as public?

### CODE REVIEW

File: src/OpenTelemetry.Api/Trace/SemanticConventions.cs [hunk 1]
```diff
@@ -30,16 +30,16 @@ public static class SemanticConventions
         public const string AttributeServiceInstance = "service.instance.id";
         public const string AttributeServiceVersion = "service.version";
 
-        public const string AttributeTelemetrySDKName = "telemetry.sdk.name";
-        public const string AttributeTelemetrySDKLanguage = "telemetry.sdk.language";
-        public const string AttributeTelemetrySDKVersion = "telemetry.sdk.version";
+        public const string AttributeTelemetrySdkName = "telemetry.sdk.name";
```
Comment (reyang) [inline src/OpenTelemetry.Api/Trace/SemanticConventions.cs:33]: Like it! Not sure why `SDK` was used initially :)

**Comment by reyang** [comment]
My take - these values should **not** be part of the public API - and this is why we call them "semantic conventions".
The conventions could evolve / change / break over time, we cannot afford to break APIs in that way.

**Comment by reyang** [comment]
In #806 we're already seeing that we have lots of "public APIs". We need to review all of them (ideally line by line) before GA to make sure we don't accidentally leak something that we'll regret later.

**Comment by CodeBlanch** [comment]
Great point @reyang. Just thinking out loud here... should we have a warning in our docs something to the effect of, our goal is to be spec-compliant which could involve breaking changes as the spec evolves? Assuming we don't have that already, I haven't caught up on all the changes.

**Comment by reyang** [comment]
@CodeBlanch currently we have very generic disclaimer [here](https://github.com/open-telemetry/opentelemetry-dotnet#release-schedule) which I guess would be okay until we arrive GA?

After GA, we don't expect any breaking changes, until we bump the major version (e.g. OpenTelemetry .NET 2.x.
I believe the spec is also following [semver](https://semver.org) so 1.1 will only have additive changes on top of 1.0 (GA), any incompatible changes will happen on 2.x or even later releases.

**Comment by CodeBlanch** [comment]
@reyang That is perfect thanks.

**Comment by alanwest** [comment]
👍 Good by me if the consensus is to address the accessibility question of all the APIs prior to GA. Slight tangent, but does there exist a general guideline that Microsoft follows for breaking changes in prerelease versions of .NET APIs?

**Comment by reyang** [comment]
> Slight tangent, but does there exist a general guideline that Microsoft follows for breaking changes in prerelease versions of .NET APIs?

Depending on what is the definition of ".NET APIs" and the definition of "Microsoft":
1. If the definition is "APIs that are expose as part of .NET runtime / framework libraries", I believe there is a general consensus although I don't know if it is clearly documented.
2. If the definition is "APIs that we expose in OpenTelemetry .NET", we are not under control of Microsoft here as the project is governed by CNCF and OpenTelemetry community, and driven by maintainers elected based on their contribution. The general guideline we follow here is coming from OpenTelemetry community.

**Comment by cijothomas** [comment]
> This PR addresses a bit of a nit, though given that the `SemanticConventions` class is currently public, I personally feel it makes sense to conform to the [design guidelines](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions#capitalization-rules-for-identifiers) for capitalization of acronyms like `Http`, `Db`, `Rpc`, `Id`, and others.
> 
> That said, having discussed this with @cijothomas, there is also the question of whether this class even ought to be public. We could make it internal, though a fair number of projects depend on this class, so would we want to be doing something like this everywhere?:
> 
> ```
> <ItemGroup>
>     <Compile Include="..\OpenTelemetry.Api\SemanticConventions.cs" />
> </ItemGroup>
> ```
> 
> Or are there better ways of dealing with this? Or do we just leave the class as public?

We can make this non public, and also share the file across all the projects - Would you do that in a future PR?

**Comment by cijothomas** [comment]
> 👍 Good by me if the consensus is to address the accessibility question of all the APIs prior to GA. Slight tangent, but does there exist a general guideline that Microsoft follows for breaking changes in prerelease versions of .NET APIs?

Noah from .NET Runtime team gave some insights here earlier - for APIs part of .NET BCL (eg Activity), there are no breaking changes ever. Between preview versions, there can be breaking changes. (an api *newly* introduced in preview1 could change in preview2 etc.). The bar to make changes after last preview is very very high. RC1 release is when .NET tells customers that we are done with API shape, so start taking dependency on it.

Activity has made all changes OpenTelemety community has asked for, and they'll come as part of the last preview, which is preview8. Additional things to API may still be possible, but any breaking change after this is difficult.

**Comment by cijothomas** [comment]
> Great point @reyang. Just thinking out loud here... should we have a warning in our docs something to the effect of, our goal is to be spec-compliant which could involve breaking changes as the spec evolves? Assuming we don't have that already, I haven't caught up on all the changes.

The blog post going out for beta is also making this explicit  - we are beta, and we expect breaking changes until we are GA.

**Pull request merged**